src/Sulu/Bundle/AdminBundle/Resources/js/containers/List/FieldFilterItem.js

```diff
@@ -0,0 +1,142 @@
+// @flow
+import React from 'react';
+import type {Node} from 'react';
+import {action, autorun, computed, observable} from 'mobx';
+import {observer} from 'mobx-react';
+import ArrowMenu from '../../components/ArrowMenu';
+import Button from '../../components/Button';
+import Chip from '../../components/Chip';
+import Loader from '../../components/Loader';
+import {translate} from '../../utils/Translator';
+import AbstractFieldFilterType from './fieldFilterTypes/AbstractFieldFilterType';
+import listFieldFilterTypeRegistry from './registries/listFieldFilterTypeRegistry';
+import fieldFilterItemStyles from './fieldFilterItem.scss';
+
+type Props = {|
+    column: string,
+    filterType: ?string,
+    filterTypeParameters: ?{[string]: mixed},
+    label: string,
+    onChange: (column: string, value: mixed) => void,
+    onClick: (column: string) => void,
+    onClose: () => void,
+    onDelete: (column: string) => void,
+    open: boolean,
+    value: mixed,
+|};
+
+@observer
+class FieldFilterItem extends React.Component<Props> {
+    @observable value: mixed;
+    fieldFilterType: AbstractFieldFilterType<*>;
+    valueDisposer: () => void;
+    valueNodeDisposer: () => void;
+    @observable valueNodeLoading: boolean = false;
+    @observable valueNode: ?Node;
+
+    constructor(props: Props) {
+        super(props);
+
+        const {filterType, filterTypeParameters, value} = this.props;
+
+        this.value = value;
+
+        if (!filterType) {
+            throw new Error(
+                'The field does not have a "filterType". This should not happen and is likely a bug.'
+            );
+        }
+
+        this.fieldFilterType = new (listFieldFilterTypeRegistry.get(filterType))(
+            this.handleFilterFieldChange,
+            filterTypeParameters,
+            value
+        );
+
+        this.valueDisposer = autorun(() => {
+            this.fieldFilterType.setValue(this.value);
+        });
+
+        this.valueNodeDisposer = autorun(() => {
+            const valueNodePromise = this.fieldFilterType.getValueNode(this.propValue);
+
+            if (valueNodePromise) {
+                this.setValueNodeLoading(true);
+                valueNodePromise.then(action((valueNode) => {
+                    this.setValueNodeLoading(false);
+                    this.setValueNode(valueNode);
+                }));
+            }
+        });
+    }
+
+    @computed get propValue() {
```

Comment:
The value passed via props is the one that currently is applied as a filter. I just need the computed getter, because otherwise I was having some mobx updating issues. I think `propValue` tells pretty good what it is. The internal value is required because the value is not passed to the outside, until the user clicks the "OK" button in the `ArrowMenu`. I don't really like `unconfirmedValue`, because the value might be confirmed, if it is not changed yet. However I could live with `displayedValue` instead of `this.value`, but I can't think of a better name for `propValue` right now. Although I am still open for suggestions :slightly_smiling_face:

Reply:
we dio use this concept of an internal `this.value` and an external `this.props.value` in other places right?

you could also use a `reaction` instead of an `autorun`, then you would not need to computed getter.

Reply:
It's just that I like reactions a lot less than `autorun`s :see_no_evil: At first, they are not executed initially by default, and I have to repeat stuff, although the `autorun` is capable of finding that out on its own.

In addition to that I think that writing `this.props.value` or `this.propValue` makes not a big difference in understanding code, if it makes a difference at all. Then I would prefer to call this getter `value` and rename the current observable to `displayedValue`.

Reply:
we are not on the same page here, i like that a reaction explicitly specifies what it reacts to. the fact that the `autorun` is executed if any of the accessed observables is changed scares me 🙂 
imo it is really easy to introduce performance problems (that are hard to spot) by accessing an observable in a method that is coincidentally called in some `autorun` (maybe even in another class).

i would not call the getter `value` because it is inconsistent with the other components that use an internal value (eg `SmartContent`). so i think we will just leave it like this 🙂

Reply:
The whole point of MobX is to avoid to explicitly specify what it has to react to, but to do it automatically instead. That's exactly why it is so great! If I want to specify every variable something reacts to, using MobX makes no sense IMO. If I change another variable that has an effect on this `autorun` I want the `autorun` to be executed in case it changes, since the result might be a different one. If I use a `reaction` instead, the possibility that I am introducing an outdated error, because I forgot a value to add to the `reaction`, is very high.

Reply:
i absolutely agree with you that this is great about mobx, but unfortunately things dont work as intuitively as they should. the whole point why you need the `propValue` getter is because you **dont** want to run the `autorun` if the `this.props` observable changes (which you would need to access if you dont hide it behind a computed getter) 😂 

somebody that does not know about this quirks will see that the `propValue` getter is used only in a single plac, inline it and cause a performance issue 🙂

Reply:
Yeah, that is a really weird behavior IMO... Still prefer that behavior over a `reaction`, as it will show bugs (if something changes that shouldn't change I'll have a chance to find it, which might otherwise cause bugs in even weirder places) and does not introduce new ones by requiring you to think about something, that MobX could actually already take care for you.